utility/src/main/java/org/oppia/android/util/profile/ProfileNameValidator.kt

```diff
@@ -0,0 +1,55 @@
+package org.oppia.android.util.profile
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Utility to validate that profile names are correctly formatted. */
+@Singleton
+class ProfileNameValidator @Inject constructor() {
+  private val noRepeatedAllowedSymbolsRegex by lazy { Regex("[.'-]{2}") }
```

Comment:
Does the '.' need to be escaped? I'm assuming that names like "Jerry" would fail right now (would be nice to add a test case for that scenario to verify a fix).

Reply:
No, the '.' doesn't need to be escaped -- Kotlin actually complains an illegal escape when I tried doing it. Also, "Jerry" passes the test, and I'm adding a test case just in case.

Reply:
Ah, did you escape it with ``\\.``? Though, if it is actually being treated as '.' correctly, then this is probably a moot point.